googleTest.cc

```diff
@@ -0,0 +1,112 @@
+#include "arm64asm.h"
+#include <cstdint>
+#include <stdlib.h>
+#include <gtest/gtest.h>
+
+// // Demonstrate some basic assertions.
+// TEST(myAbs, BasicAssertions) {
+//   // Expect two strings not to be equal.
+
+//   EXPECT_EQ(myAbs(-8), 8);
+//   EXPECT_EQ(myAbs(-4294967296), 4294967296);
+//   EXPECT_EQ(myAbs(8), 8);
+//   // Expect equality.
+//   EXPECT_EQ(myAbs(0), 0);
+//   EXPECT_EQ(myAbs(4294967296), 4294967296);
+// }
+
+// bool AssertMemset(void *p, int count, 
+// uint8_t pattern) {
+//     uint8_t *buf = (uint8_t*)p;
+//     for (int i = 0; i < count; i ++) {
+//         if (buf[i] != pattern) {
+//             return false;
+//         }
+//     }
+//     return true;
+// }
+
+// TEST(myMemset, BasicAssertions) {
+//     void *buffer = malloc(1024*1024*32);
+//     EXPECT_EQ(myMemset(buffer, 8, 0), 8);
+//     EXPECT_PRED3(AssertMemset, buffer, 8, 0);
+//     EXPECT_EQ(myMemset(buffer, 1024*1024*32, 0), 1024*1024*32);
+//     EXPECT_PRED3(AssertMemset, buffer, 1024*1024*32, 0);
+//     free(buffer);
+// }
+
+void PrintArray(int32_t *array, int counts) {
+    for (int i = 0; i < counts; i ++) {
+        printf("%d ", array[i]);
+    }
+    printf("\n");
+}
+
+void PrintArray2(int32_t *array, int begin, int end) {
+    for (int i = begin; i <= end; i ++) {
+        printf("%d ", array[i]);
+    }
+    printf("\n");
+}
+
+bool AssertSort(int32_t *array, int counts) {
+    int32_t temp = array[0];
+    PrintArray(array, counts);
+    for (int i = 1; i < counts; i ++) {
+        if (array[i] < temp) {
+            printf("assert failed, arr[%d]=%d < arr[%d] = %d\n", i, array[i], i - 1, temp);
+            return false;
+        }
+        temp = array[i];
+    }
+    return true;
+}
+
+void QuickSort(int32_t *array, int begin, int end) {
+    if (begin >= end) {
+        return;
+    }
+    int32_t pivot = partition(array, begin, end);
+    printf("pivot=%d,\t", pivot);
+    PrintArray2(array, begin, end);
+    QuickSort(array, begin, pivot - 1);
+    QuickSort(array, pivot + 1, end);
+}
```

Comment:
All qsort function details should implement with arm64 assembly. In test, the caller only call single function and get result. That means, the function QuickSort logic should also assembly written.